Drop the 1st measurement as inaccurate

hyperslv opened the pull request:
I noticed that the very first reading after my board is powered on is inaccurate and differs significantly from next ones:
```
iter#000: Temp = 30.250 deg C; rH = 61.5%  
iter#001: Temp = 25.500 deg C; rH = 61.0%   
iter#002: Temp = 25.500 deg C; rH = 61.0%
```
So my PR is to drop the very first reading at the end of initialization and retrigger new measurement if the output data rate mode is set to 'one shot'.

hyperslv committed:
Drop 1st measurement as inaccurate and retrigger new measurement for one shot mode

danielgallagher0 commented:
It's been a while since I've run my demo, but I didn't see any issues that I recall. Are you checking the status flags before taking the measurement?

hyperslv commented:
> It's been a while since I've run my demo, but I didn't see any issues that I recall. Are you checking the status flags before taking the measurement?

Yes, sure. There is my code sample:
`
#![no_std]
#![no_main]
#![deny(unsafe_code)]

use core::num::Wrapping;

// pick a panicking behavior
//use panic_halt as _; // you can put a breakpoint on `rust_begin_unwind` to catch panics
use panic_semihosting as _; // logs messages to the host stderr; requires a debugger

use cortex_m_rt::entry;
use hts221::device;
use stm32f3xx_hal::{delay::Delay, prelude::*, stm32};
use switch_hal::{IntoSwitch, ToggleableOutputSwitch};

use cortex_m_semihosting::hprintln;

#[entry]
fn main() -> ! {
    let core = cortex_m::Peripherals::take().unwrap();
    let chip = stm32::Peripherals::take().unwrap();

    // Configure clocks
    let mut rcc = chip.RCC.constrain();
    let mut flash = chip.FLASH.constrain();
    let clocks = rcc.cfgr.freeze(&mut flash.acr);

    // Configure the Sys Timer for Delay
    let mut delay = Delay::new(core.SYST, clocks);

    // User LED = PA5, high - LED ON
    let mut gpioa = chip.GPIOA.split(&mut rcc.ahb);
    let mut led = gpioa
        .pa5
        .into_push_pull_output(&mut gpioa.moder, &mut gpioa.otyper)
        .downgrade()
        .into_active_high_switch();

    // I2C pins to NUCLEO-IKS01A3: I2C1_SCL - PB8, I2C1_SDA - PB9, INT - ?
    let mut gpiob = chip.GPIOB.split(&mut rcc.ahb);
    let scl = gpiob.pb8.into_af4(&mut gpiob.moder, &mut gpiob.afrh);
    let sda = gpiob.pb9.into_af4(&mut gpiob.moder, &mut gpiob.afrh);
    let mut i2c =
        stm32f3xx_hal::i2c::I2c::i2c1(chip.I2C1, (scl, sda), 50.khz(), clocks, &mut rcc.apb1);

    // Create a HTS221 instance
    let mut hts221 = hts221::Builder::new()
        .with_avg_t(hts221::AvgT::Avg256)
        .with_avg_h(hts221::AvgH::Avg512)
        .build(&mut i2c)
        .unwrap();

    hprintln!("Hello, slv!").unwrap();
    hprintln!("Let's print temperature and humidity from the HTS221").unwrap();

    let mut i = Wrapping(0u8);
    loop {
        led.toggle().ok();

        // check data availability from the HTS221
        loop {
            match hts221.status(&mut i2c) {
                Ok(status) => {
                    if status.humidity_data_available() && status.temperature_data_available() {
                        break;
                    }
                }
                Err(_) => hprintln!("Could not get status").unwrap(),
            }
        }

        // read temperature and humidity
        let temperature_x8 = hts221.temperature_x8(&mut i2c).unwrap();
        let humidity_x2 = hts221.humidity_x2(&mut i2c).unwrap();

        // Print
        hprintln!(
            "iter#{:03}: Temp = {}.{} deg C; rH = {}.{}%",
            i,
            temperature_x8 >> 3,
            125 * (temperature_x8 & 0b111),
            humidity_x2 >> 1,
            5 * (humidity_x2 & 0b1)
        )
        .unwrap();

        // trigger next measurement
        let mut hts221_device = device::Device::new(device::I2C_ID_7BIT, &mut i2c);
        let mut cr2 = device::CtrlReg2::new(&mut hts221_device).unwrap();
        cr2.modify(&mut hts221_device, |w| w.set_one_shot()).unwrap();

        delay.delay_ms(3000u32); // has to be less than 4000 ms for the clock setting
        i += Wrapping(1u8);
    }
}
`
